pkg/install/2-removebootstrap.go

```diff
@@ -135,6 +141,107 @@ func (i *Installer) removeBootstrap(ctx context.Context) error {
 		return err
 	}
 
+	if _, ok := i.env.(env.Dev); !ok &&
+		i.env.ManagedDomain(i.doc.OpenShiftCluster.Properties.ClusterDomain) != "" {
+		kcli, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		for _, c := range []struct {
+			certificateName string
+			namespace       string
+		}{
+			{
+				certificateName: i.doc.ID + "-apiserver",
+				namespace:       "openshift-config",
+			},
+			{
+				certificateName: i.doc.ID + "-ingress",
+				namespace:       "openshift-ingress",
+			},
+		} {
+			key, certs, err := i.keyvault.GetSecret(ctx, c.certificateName)
+			if err != nil {
+				return err
+			}
+
+			b, err := x509.MarshalPKCS8PrivateKey(key)
+			if err != nil {
+				return err
+			}
+
+			err = kcli.CoreV1().Secrets(c.namespace).Delete(c.certificateName, &metav1.DeleteOptions{})
+			if err != nil && !errors.IsNotFound(err) {
+				return err
+			}
+
+			_, err = kcli.CoreV1().Secrets(c.namespace).Create(&v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: c.certificateName,
+				},
+				Data: map[string][]byte{
+					v1.TLSCertKey:       pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certs[0].Raw}),
+					v1.TLSPrivateKeyKey: pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: b}),
+				},
+				Type: v1.SecretTypeTLS,
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		ocli, err := operatorclient.NewForConfig(restConfig)
+		if err != nil {
+			return err
+		}
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			ic, err := ocli.OperatorV1().IngressControllers("openshift-ingress-operator").Get("default", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			ic.Spec.DefaultCertificate = &v1.LocalObjectReference{
+				Name: i.doc.ID + "-apiserver",
+			}
+
+			_, err = ocli.OperatorV1().IngressControllers("openshift-ingress-operator").Update(ic)
+			return err
+		})
+		if err != nil {
+			return err
+		}
+
+		// TODO: wait
```

Comment:
I think we should aggregate all objects and have one wait loop below with `map[object]bool` so update would be sent to API and waits are parallelized.

Reply:
+1.